TAP formatter only display one message per file.

Version :
`1.10.3`

File to lint : 
```
var a = 'foo';
var b = 'bar';
```

Configuration : 

```
{
  "extends": "airbnb"
}
```

Result with default formatter: 

```
/my/path/test.js
  1:1  error  Unexpected var, use let or const instead  no-var
  1:5  error  "a" is defined but never used             no-unused-vars
  2:1  error  Unexpected var, use let or const instead  no-var
  2:5  error  "b" is defined but never used             no-unused-vars

✖ 4 problems (4 errors, 0 warnings)
```

Result with TAP formatter: 

```
TAP version 13
1..1
not ok 1 - /my/path/test.js
  ---
  message: 'Unexpected var, use let or const instead.'
  severity: error
  data:
    line: 1
    column: 1
    ruleId: no-var
  messages:
    - message: '"b" is defined but never used'
      severity: error
      data:
        line: 2
        column: 5
        ruleId: no-unused-vars
  ...
```

Problem: 

The TAP formatter swallow messages and only display the first and the last one.



<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/29931508-tap-formatter-only-display-one-message-per-file?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/lib/formatters/tap.js
+++ b/lib/formatters/tap.js
@@ -66,7 +66,10 @@ module.exports = function(results) {
                 // The first error will be logged as message key
                 // This is to adhere to TAP 13 loosely defined specification of having a message key
                 if ("message" in diagnostics) {
-                    diagnostics.messages = [diagnostic];
+                    if (typeof diagnostics.messages === "undefined") {
+                        diagnostics.messages = [];
+                    }
+                    diagnostics.messages.push(diagnostic);
                 } else {
                     diagnostics = diagnostic;
                 }

--- a/tests/lib/formatters/tap.js
+++ b/tests/lib/formatters/tap.js
@@ -91,6 +91,12 @@ describe("formatter:tap", function() {
                 line: 6,
                 column: 11,
                 ruleId: "bar"
+            }, {
+                message: "Unexpected baz.",
+                severity: 1,
+                line: 7,
+                column: 12,
+                ruleId: "baz"
             }]
         }];
 
@@ -104,6 +110,9 @@ describe("formatter:tap", function() {
             assert.include(result, "Unexpected bar.");
             assert.include(result, "line: 6");
             assert.include(result, "column: 11");
+            assert.include(result, "Unexpected baz.");
+            assert.include(result, "line: 7");
+            assert.include(result, "column: 12");
         });
     });
 

